server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -220,7 +246,21 @@ private <Params extends PersistentTaskParams> void doStartTask(
                     + "], removing from persistent tasks",
                 e
             );
-            notifyMasterOfFailedTask(taskInProgress, e);
+
+            // create a no-op placeholder task so that we don't keep trying to start this task while we wait for the cluster state update
+            // which handles the failure
+            final var placeholderTask = (AllocatedPersistentTask) taskManager.register(
```

Comment:
Can this register fail too?

Reply:
I don't see how, the creation of this task doesn't do anything interesting enough to fail AFAICT.